packages/sdk/src/browser-wallets/okx/index.ts

```diff
@@ -0,0 +1,200 @@
+import { OKXSignPSBTOptions } from "./types";
+
+type OKXError = { code?: number; message: string };
+
+function isInstalled(): boolean {
+  if (typeof window === "undefined") {
+    throw new OrditSDKError("Cannot call this function outside a browser");
+  }
+  return typeof window.okxwallet !== "undefined";
+}
+
+function getOKXWalletProvider(
+  network: BrowserWalletNetwork = "mainnet",
+): OKXWalletProvider {
+  if (!isInstalled()) {
+    throw new BrowserWalletNotInstalledError("OKX Wallet not installed.");
+  }
+
+  const provider =
+    network === "mainnet"
+      ? window.okxwallet.bitcoin
+      : window.okxwallet.bitcoinTestnet;
+
+  if (!provider) {
+    throw new OrditSDKError("Failed to get OKX Wallet provider.");
+  }
+
+  return provider;
+}
+
+async function getAddresses(
+  network: BrowserWalletNetwork = "mainnet",
+): Promise<WalletAddress[]> {
+  if (!isInstalled()) {
+    throw new BrowserWalletNotInstalledError("OKX Wallet not installed.");
+  }
+
+  const provider = getOKXWalletProvider(network);
+
+  try {
+    const { address, publicKey } = await provider.connect();
+    const format = getAddressFormat(address, network);
+
+    const fullPubKey = fromXOnlyToFullPubkey(publicKey);
+
+    if (!address || !fullPubKey || !format) {
+      throw new OrditSDKError("Failed to get addresses from OKX Wallet.");
+    }
+
+    return [
+      {
+        publicKey: fullPubKey,
+        address,
+        format,
+      },
+    ];
+  } catch (err) {
+    if (err instanceof OrditSDKError) {
+      // internal error caused by getAddressFormat
+      throw err;
+    }
+    const okxError = err as OKXError;
+    if (okxError.code === 4001) {
+      throw new BrowserWalletRequestCancelledByUserError();
+    }
+    throw new OrditSDKError(okxError.message);
+  }
+}
+
+async function signPsbt(
+  psbt: Psbt,
+  {
+    finalize = true,
+    extractTx = true,
+    network,
+    inputsToSign,
+  }: OKXSignPSBTOptions = { network: "mainnet", inputsToSign: [] },
+): Promise<BrowserWalletSignResponse> {
+  if (!isInstalled()) {
+    throw new BrowserWalletNotInstalledError("OKX Wallet not installed.");
+  }
+
+  if (extractTx && !finalize) {
+    throw new BrowserWalletExtractTxFromNonFinalizedPsbtError();
+  }
+
+  const provider = getOKXWalletProvider(network);
+
+  const psbtHex = psbt.toHex();
+
+  let signedPsbtHex: string = "";
+
+  const toSignInputs: OKXSignInput[] = [];
+
+  inputsToSign.forEach((input) => {
+    const { address, signingIndexes } = input;
+    signingIndexes.forEach((index) => {
+      toSignInputs.push({
+        index,
+        address,
+      });
+    });
+  });
+
+  try {
+    signedPsbtHex = await provider.signPsbt(psbtHex, {
+      autoFinalized: finalize,
+      toSignInputs,
+    });
+  } catch (err) {
+    const okxError = err as OKXError;
+    if (okxError.code === 4001) {
+      throw new BrowserWalletRequestCancelledByUserError();
+    }
+    throw new OrditSDKError(okxError.message);
```

Comment:
Should we propagate the okx error? If it's empty, should we let it fall through to line 131?